src/item.cpp

```diff
@@ -6852,21 +6873,37 @@ bool item::craft_has_charges()
 #pragma optimize( "", off )
 #endif
 
-float item::bash_resist( bool to_self ) const
+float item::bash_resist( bool to_self, const bodypart_id &bp ) const
 {
     if( is_null() ) {
         return 0.0f;
     }
+    const bool bp_null = bp == bodypart_id();
 
     float resist = 0.0f;
     float mod = get_clothing_mod_val( clothing_mod_type_bash );
-
-    // base resistance
-    // Don't give reinforced items +armor, just more resistance to ripping
     const int dmg = damage_level();
     const float eff_damage = to_self ? std::min( dmg, 0 ) : std::max( dmg, 0 );
-    const float eff_thickness = std::max( 0.1f, get_thickness() - eff_damage );
 
+    if( !bp_null ) {
+        const std::vector<const part_material *> &armor_mats = armor_made_of( bp );
+        // If we have armour portion materials for this body part, use that instead
+        if( !armor_mats.empty() ) {
+            int total = 0;
+            for( const part_material *m : armor_mats ) {
+                const float eff_thic = std::max( 0.1f, m->thickness - eff_damage );
+                resist += m->id->bash_resist() * m->portion * eff_thic;
+                total += m->portion;
+            }
+            resist /= total;
```

Comment:
If for physical properties (bash, cut, ranged) you didn't divide by the total
and each portion for each material was capped between 0-1 

So an elbow pad could be 100% nylon (portion 1) but only 20% plastic (portion .2) that would mean you could specify the correct thickness values but portions would still work correctly for acid, fire etc.